text/0046-ttfd-automatic-transaction-span.md

```diff
@@ -0,0 +1,101 @@
+- Start Date: 2022-12-16
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/46
+- RFC Status: draft
+  
+# Summary. 
+  
+We want to provide a new span to the automatic UI transactions.  
+The TTFD (time-to-full-display) span is a way for the user to notify the SDK that the UI has been fully loaded.  
+That is, after all data is retrieved, either by database calls or rest APIs, and set into the UI.  
+We would create a new span to the UI automatic transactions to measure it, for all screens of the application.  
+This rfc is mostly about how we should design the new API.  
+  
+# Motivation
+
+This is useful especially for mobile developers, but could also be useful for web.  
+There is no reliable way to automatically detect when UI is fully drawn, as the "fully drawn" concept depends on the developer.  
+  
+# Background. 
+  
+We have to add a new API to the SDK to allow the user to notify that the UI was fully drawn.  
+Also, we need the user to specify the span to finish through a parameter, as it cannot be done automatically.  
+E.g. Activity A starts -> Activity B starts -> Activity A finishes loading data and the API is called.  
+At this point without the activity it was called on, we wouldn't know which span to finish, because the activity B would be at the top of the stack.  
+  
+# Options Considered
+  
+These options were considered for Android, but the same apply to other SDKs, too.  
+1) Add a `SentryAndroid.reportFullyDrawn(Activity)` static method. We would start the span automatically when an Activity is being created and we would finish it when the API is called.  
```

Comment:
`reportFullyDrawn` is an Android API, if our feature is called TTFD, I'd suggest naming it to something similar.
I'd also try to add this API to  `SentryAndroid`, also if possible, to not require `Activity`, so it can't be Android less classes.
Why do we need the activity?

Reply:
> Im not sure about that, it depends what they call app start cold/hot/warm state, maybe they consider app start just by bringing the app from the background which is not even the 1st screen anymore, so it'd be fine to be called or it will be ignored if already, hard to tell without reading the source code/more docs.

I tried to call `Activity.reportFullyDrawn()` multiple times in multiple activities, and it gets called once per activity, so it would be fine. There is still the question about single activity apps, tough. However, i updated the rfc.

the name would be `time-to-full-display`. The activity is needed (in some options) because we wouldn't know which activity is the one that was fully drawn, in case two activities are started and the previous one is not destroyed.